code-challenges.test.js

```diff
@@ -14,48 +14,134 @@
+       var arrayOfWords1 = ["Apple", "Banana", "Plum", "Orange", "Kiwi"]
+       // Expected output: ["Apple", "Banana", "Orange"]
+       expect(findingLetter(arrayOfWords1)).toEqual(["Apple", "Banana", "Orange"])
+       var arrayOfWords2 = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach"]
+       // Expected output: ["Mango", "Apricot", "Peach"]
+       expect(findingLetter(arrayOfWords2)).toEqual(["Mango", "Apricot", "Peach"])
+    })
+})
 // b) Create the function that makes the test pass.
+const findingLetter = (arr) => {
+    let containsLetterA = [];
+    arr.forEach(word => {
+        if(word.toLowerCase().indexOf('a') !== -1) {
+            containsLetterA.push(word);
+        }
+    });
+
+    return containsLetterA;
+}
 
 
 
 // --------------------3) Create a function that takes in an array of 5 numbers and determines whether or not the array is a “full house”. A full house is exactly one pair and one three of a kind.
+/////////////////////////////////////////////////// Todd's Pseudo Code ///////////////////////////////////////////////////
+// this one give me fits and took me a long time to mentally process what I needed to do.
+// I tried breaking it down into two different array's and then evaluating those array's the first array was 
+// three and the second array was two then return true if not return false.
+// I think my idea is correct but the logic is what I was having a hard time with it.
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
 // a) Create a test with an expect statement using the variable provided.
-
-var hand1 = [5, 5, 5, 3, 3]
-// Expected output: true
-var hand2 = [5, 5, 3, 3, 4]
-// Expected output: false
-var hand3 = [5, 5, 5, 5, 4]
-// Expected output: false
-
-
-
+describe("fullhouse", () => {
+    test("takes in an array of 5 numbers and determines whether or not the array has 2 of the numbers and 3 of the same numbers", () => {
+       var hand1 = [5, 5, 5, 3, 3]
+       // Expected output: true
+       expect(fullhouse(hand1)).toEqual("true")
+       var hand2 = [5, 5, 3, 3, 4]
+       // Expected output: false
+       expect(fullhouse(hand2)).toEqual("false")
+       var hand3 = [5, 5, 5, 5, 4]
+       // Expected output: false
+       expect(fullhouse(hand3)).toEqual("false")
+    })
+})
 // b) Create the function that makes the test pass.
+
+const fullHouse = (arr) => {
+    let firstArr = [];
+    let secondArr = [];
+    
+    for(let i = 0; i < arr.length; i++){
+      if(i === 0) {
+         firstArr.push(arr[i]);
+      }else if(arr[i] === firstArr[0]) {
+        firstArr.push(arr[i]);
+      }else if(secondArr.length === 0) {
+        secondArr.push(arr[i]);
+      }else if(arr[i] === secondArr[0]) {
+        secondArr.push(arr[i]);
+      }
+    }
+    
+    if(firstArr.length === 2 && secondArr.length === 3 || firstArr.length === 3 && secondArr.length === 2) {
```

Comment:
I get it! this is creative! I can see where your going with this. I think keeping a couple counters instead of arrays would make life easier. Maybe it wouldn't be a whole lot different.